### ISSUE TITLE
Are the released pre-trained models the best models?

### ISSUE THREAD

**Issue by intellisense-team** [sshan-zhao/ACMNet#7]
Hi! Thanks for the code and the pretrained model.
I have tested the pretrained model on the validation set, but I couldn't get the same results as the paper said. The results are as follow.
![image](https://user-images.githubusercontent.com/65592911/127604585-8256379b-8057-4411-a562-8e78f5979ef8.png)

Besides, I trained the model for about 40 epochs and retested the validation set, but only got the results as follow.
![image](https://user-images.githubusercontent.com/65592911/127604786-89de36a3-c0d2-449e-931c-4c7d915f381f.png)

So, how can I get the results as the paper described? Thanks!

**Comment by sshan-zhao** [comment]
> Hi! Thanks for the code and the pretrained model.
> I have tested the pretrained model on the validation set, but I couldn't get the same results as the paper said. The results are as follow.
> ![image](https://user-images.githubusercontent.com/65592911/127604585-8256379b-8057-4411-a562-8e78f5979ef8.png)
> 
> Besides, I trained the model for about 40 epochs and retested the validation set, but only got the results as follow.
> ![image](https://user-images.githubusercontent.com/65592911/127604786-89de36a3-c0d2-449e-931c-4c7d915f381f.png)
> 
> So, how can I get the results as the paper described? Thanks!

The RMSE is 1047? How did you evaluate the model? The results are wrong.

**Comment by intellisense-team** [comment]
I just built the same environment and run the script "python val.py --model test --channels 64 --model_path model_64.pth --clip --knn 6 6 6 --nsamples 10000 5000 2500 --flip_input".

**Comment by intellisense-team** [comment]
> > Hi! Thanks for the code and the pretrained model.
> > I have tested the pretrained model on the validation set, but I couldn't get the same results as the paper said. The results are as follow.
> > ![image](https://user-images.githubusercontent.com/65592911/127604585-8256379b-8057-4411-a562-8e78f5979ef8.png)
> > Besides, I trained the model for about 40 epochs and retested the validation set, but only got the results as follow.
> > ![image](https://user-images.githubusercontent.com/65592911/127604786-89de36a3-c0d2-449e-931c-4c7d915f381f.png)
> > So, how can I get the results as the paper described? Thanks!
> 
> The RMSE is 1047? How did you evaluate the model? The results are wrong.

I just built the same environment and run the script "python val.py --model test --channels 64 --model_path model_64.pth --clip --knn 6 6 6 --nsamples 10000 5000 2500 --flip_input".

**Comment by sshan-zhao** [comment]
> > > Hi! Thanks for the code and the pretrained model.
> > > I have tested the pretrained model on the validation set, but I couldn't get the same results as the paper said. The results are as follow.
> > > ![image](https://user-images.githubusercontent.com/65592911/127604585-8256379b-8057-4411-a562-8e78f5979ef8.png)
> > > Besides, I trained the model for about 40 epochs and retested the validation set, but only got the results as follow.
> > > ![image](https://user-images.githubusercontent.com/65592911/127604786-89de36a3-c0d2-449e-931c-4c7d915f381f.png)
> > > So, how can I get the results as the paper described? Thanks!
> > 
> > 
> > The RMSE is 1047? How did you evaluate the model? The results are wrong.
> 
> I just built the same environment and run the script "python val.py --model test --channels 64 --model_path model_64.pth --clip --knn 6 6 6 --nsamples 10000 5000 2500 --flip_input".

Hi,
I just evaluate the model again using the code in my machine (should be the same as the repository), and get the results:  
205.35173 762.745 0.8842766 2.1444786 0.997923 0.99935645 0.9997207 0.011058769
Did you modify the code?

**Comment by sshan-zhao** [comment]
> > > Hi! Thanks for the code and the pretrained model.
> > > I have tested the pretrained model on the validation set, but I couldn't get the same results as the paper said. The results are as follow.
> > > ![image](https://user-images.githubusercontent.com/65592911/127604585-8256379b-8057-4411-a562-8e78f5979ef8.png)
> > > Besides, I trained the model for about 40 epochs and retested the validation set, but only got the results as follow.
> > > ![image](https://user-images.githubusercontent.com/65592911/127604786-89de36a3-c0d2-449e-931c-4c7d915f381f.png)
> > > So, how can I get the results as the paper described? Thanks!
> > 
> > 
> > The RMSE is 1047? How did you evaluate the model? The results are wrong.
> 
> I just built the same environment and run the script "python val.py --model test --channels 64 --model_path model_64.pth --clip --knn 6 6 6 --nsamples 10000 5000 2500 --flip_input".

In addition, which version of pytorch do you use? Sometimes, different functions are used in different versions of pytorch.

**Comment by intellisense-team** [comment]
Hi! I just re-run the evaluate script. All the things are not changed. But this time I get the result:
![image](https://user-images.githubusercontent.com/65592911/127653013-89c77b7e-ad9e-4f33-afef-38b675189607.png)
It makes me confused.
BTW, I didn't modify the code, and the version of pytorch is 1.2. The only thing different from the README is that the version of ubuntu is 18.04.

**Comment by intellisense-team** [comment]
> Hi! I just re-run the evaluate script. All the things are not changed. But this time I get the result:
> ![image](https://user-images.githubusercontent.com/65592911/127653013-89c77b7e-ad9e-4f33-afef-38b675189607.png)
> It makes me confused.
> BTW, I didn't modify the code, and the version of pytorch is 1.2. The only thing different from the README is that the version of ubuntu is 18.04.

@sshan-zhao

**Comment by sshan-zhao** [comment]
> > Hi! I just re-run the evaluate script. All the things are not changed. But this time I get the result:
> > ![image](https://user-images.githubusercontent.com/65592911/127653013-89c77b7e-ad9e-4f33-afef-38b675189607.png)
> > It makes me confused.
> > BTW, I didn't modify the code, and the version of pytorch is 1.2. The only thing different from the README is that the version of ubuntu is 18.04.
> 
> @sshan-zhao

The method needs to sample 10000, 5000, 2500 points randomly, which might result in the issue. I guess. I never have such issue.

**Comment by intellisense-team** [comment]
> > > Hi! I just re-run the evaluate script. All the things are not changed. But this time I get the result:
> > > ![image](https://user-images.githubusercontent.com/65592911/127653013-89c77b7e-ad9e-4f33-afef-38b675189607.png)
> > > It makes me confused.
> > > BTW, I didn't modify the code, and the version of pytorch is 1.2. The only thing different from the README is that the version of ubuntu is 18.04.
> > 
> > 
> > @sshan-zhao
> 
> The method needs to sample 10000, 5000, 2500 points randomly, which might result in the issue. I guess. I never have such issue.

Thanks!!!
